railties/test/application/credentials_test.rb

```diff
@@ -11,46 +11,54 @@ class Rails::CredentialsTest < ActiveSupport::TestCase
 
   test "reads credentials from environment specific path" do
     write_credentials_override(:production)
-
     app("production")
-
     assert_equal "revealed", Rails.application.credentials.mystery
   end
 
   test "reads credentials from customized path and key" do
     write_credentials_override(:staging)
     add_to_env_config("production", "config.credentials.content_path = config.root.join('config/credentials/staging.yml.enc')")
     add_to_env_config("production", "config.credentials.key_path = config.root.join('config/credentials/staging.key')")
-
     app("production")
-
     assert_equal "revealed", Rails.application.credentials.mystery
   end
 
   test "reads credentials using environment variable key" do
     write_credentials_override(:production, with_key: false)
-
     switch_env("RAILS_MASTER_KEY", credentials_key) do
       app("production")
-
       assert_equal "revealed", Rails.application.credentials.mystery
     end
   end
 
-  private
-    def write_credentials_override(name, with_key: true)
-      Dir.chdir(app_path) do
-        Dir.mkdir  "config/credentials"
-        File.write "config/credentials/#{name}.key", credentials_key if with_key
-
-        # secret_key_base: secret
-        # mystery: revealed
-        File.write "config/credentials/#{name}.yml.enc",
-          "vgvKu4MBepIgZ5VHQMMPwnQNsLlWD9LKmJHu3UA/8yj6x+3fNhz3DwL9brX7UA==--qLdxHP6e34xeTAiI--nrcAsleXuo9NqiEuhntAhw=="
+  test "application has convenience method for accessing credentials" do
+    app("production")
+    assert_equal(Rails.application.credentials, Rails.credentials)
```

Comment:
The test is failing with `undefined method `credentials' for Rails:Module` and the failure seems legit as I can't find definition or delegation for this method. Is there a documentation entry that implies that such method exists?

Reply:
We could add convenience method for accessing credentials similar to "Rails.credentials" soemthing like "config/initializers/credentials.rb"

for example:
------------------
```
module Rails
  def self.credentials
    application.credentials
  end
end
```
------------------
This will define a Rails.credentials method that delegates to "Rails.application.credentials", which should allow your test case to pass.
Or we could close the pull request, and I apologize for wasting your times.

Reply:
I think we should separate the two, as it's unclear whether that PR will be accepted and merged.

Would you mind rebasing those changes out? :bow:

Reply:
yes please rebasing those changes